### PR TITLE
feat: add auto generate the meaning title for the message using AI

### LINKED ISSUE #92: Auto generate the meaning title for the message use the AI
in the frontend/lib/tools/auto.ts

You can initiate a request to gpt-4o-mini to summarize user questions in parallel.

```
import { generateText } from 'ai';

const { text } = await generateText({
  model: yourModel,
  system:
    'You are a professional writer. ' +
    'You write simple, clear, and concise content.',
  prompt: `Summarize the following article in 3-5 sentences: ${article}`,
});
```

The summary title and answer can then be streamed back to the client

```
   await streamResponse({ sources: texts, title: title }, onStream);
```

Finally, when savingMessages, also pass in the AI ​​generated title

```
      await saveMessages(userId, messages, fullAnswer, texts, images, videos, fullRelated);
```

### PR DESCRIPTION
**Pull request by krushnarout**
## Description

Added auto generate the meaning title for the message using gpt-4o-mini.

## Related Issue

- Closes #92 

## Changes Made

- [x] Added new feature
- [ ] Fixed a bug
- [ ] Updated documentation
- [ ] Other (please specify)

## How to Test

Please describe the steps to test the changes made in this pull request. Include any specific commands or configurations needed.

1. Step 1
2. Step 2
3. Step 3

## Checklist

- [x] I have read the contributing guidelines
- [x] I have followed the code style guidelines
- [x] My code is tested and works as expected
- [ ] I have updated the documentation (if necessary)

## Additional Information

Please provide any additional information that may be helpful for the reviewers.

### CODE REVIEW

File: frontend/lib/tools/auto.ts [hunk 1]
```diff
@@ -30,6 +31,21 @@ export async function autoAnswer(
         const newMessages = getHistoryMessages(isPro, messages);
         const query = newMessages[newMessages.length - 1].content;
 
+        let summaryTitle = '';
+        let summaryText = '';
+
+        try {
+            const { text } = await generateText({
```
Comment (ahaapple) [inline frontend/lib/tools/auto.ts:38]: No need for await, we should prioritize streamText, and we can await generateText after streamText ends.

File: frontend/lib/tools/auto.ts [hunk 2]
```diff
@@ -30,6 +31,21 @@ export async function autoAnswer(
...
+                model: getLLM('gpt-4o-mini'),
+                system: 'You are a professional writer. You write simple, clear, and concise content.',
+                prompt: `Summarize the following article in 3-5 sentences: ${query}`,
```
Comment (ahaapple) [inline frontend/lib/tools/auto.ts:41]: Please give the following question a concise title

File: frontend/lib/tools/auto.ts [hunk 3]
```diff
@@ -173,6 +189,11 @@ export async function autoAnswer(
             await streamResponse({ videos: videos }, onStream);
         }
 
+        if (summaryText) {
+            await streamResponse({ sources: texts, title: summaryTitle }, onStream);
```
Comment (ahaapple) [inline frontend/lib/tools/auto.ts:193]: we should update the title in the UI

File: frontend/lib/tools/auto.ts (hunk 2 above)
Comment (krushnarout) [reply frontend/lib/tools/auto.ts]: Got it!

**Comment by krushnarout** [comment]
Hi @ahaapple, I've added changes according to your comments, please review and let me know!

**Comment by ahaapple** [comment]
This PR only completes part of the functionality. I will merge it first, and then I will complete the rest of the development. Thank you.

**Comment by krushnarout** [comment]
> This PR only completes part of the functionality. I will merge it first, and then I will complete the rest of the development. Thank you.

Thanks for merging!

**Pull request merged**